src/golang/lib/lib_utils/utils.go

```diff
@@ -84,21 +84,33 @@ func RunCmd(command string, args []string, dir string, stream bool) (string, str
 		stdoutScanner := bufio.NewScanner(stdout)
 		stderrScanner := bufio.NewScanner(stderr)
 
+		var stderrMsg string
+		// Create a channel to communicate between the main process and the goroutine to exchange
+		// the stderr message.
+		ch := make(chan string)
+
 		// start separate goroutines to stream the output from each scanner
 		go func() {
 			for stdoutScanner.Scan() {
 				log.Infof("stdout: %s", stdoutScanner.Text())
 			}
 		}()
 		go func() {
+			var sb strings.Builder
 			for stderrScanner.Scan() {
 				log.Errorf("stderr: %s", stderrScanner.Text())
+				sb.WriteString(stderrScanner.Text())
+				sb.WriteString("\n")
 			}
+			ch <- sb.String()
 		}()
 
-		// wait for the command to complete
+		// Wait for the stderr goroutine to finish and receive the stderr from the channel.
+		stderrMsg = <-ch
```

Comment:
hmm I would expect this to block forever until something is written to ch... is something guaranteed to be written?

Reply:
and yes you can send an empty string to channel and the listener blocked on the channel will get unblocked.